presto-parser/src/main/java/com/facebook/presto/sql/tree/ExpressionTreeRewriter.java

```diff
@@ -855,6 +855,19 @@ protected Expression visitQuantifiedComparisonExpression(QuantifiedComparisonExp
 
             return node;
         }
+
+        @Override
+        public Expression visitGroupingOperation(GroupingOperation node, Context<C> context)
```

Comment:
For `ExpressionExtractor` that visitor traverses the `PlanNode`s so adding a visitor for `GroupingOperation` doesn't make sense. Also any `GroupingOperations` would be extracted in `visitProject()`.

For `DesugaringOptimizer` I added a check to the `desugar()` method because that is also a visitor of `PlanNode`s and not expressions.

Let me know if I've misunderstood you somehow.